### PR TITLE
Drop support for Ruby 2.3 and older 

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@ekohl fixed it

File: Gemfile [hunk 1]
```diff
@@ -3,43 +3,20 @@ source 'https://rubygems.org'
 gemspec
 
 group :test do
-  gem 'rake', '~> 10.0'
+  gem 'rake', '>= 10.0'
```
Comment (ekohl) [inline Gemfile:6]: I think I'd pin it to 13
```suggestion
  gem 'rake', '~> 13.0'
```

**Review by ekohl — COMMENTED** [review]
I'm a bit unsure about `%i[]`. Personally I don't like multiple ways of writing the same thing and I don't write enough Ruby to remember `%i[arrow_alignment hard_tabs]` should be read as `[:arrow_alignment, :hard_tabs]`. Maybe this is my Python mind, but I like it when there's one way to write things.

File: lib/puppet-lint/plugins/check_whitespace/arrow_alignment.rb [hunk 2]
```diff
@@ -103,7 +103,7 @@ def fix(problem)
       problem[:token].prev_code_token.prev_token.value = ' ' * problem[:newline_indent]
 
       end_param_idx = tokens.index(problem[:token].prev_code_token)
-      start_param_idx = tokens.index(problem[:token].prev_token_of([:INDENT, :NEWLINE]))
+      start_param_idx = tokens.index(problem[:token].prev_token_of(%i[INDENT NEWLINE]))
       param_length = tokens[start_param_idx..end_param_idx].map { |r| r.to_manifest.length }.reduce(0) { |sum, x| sum + x } + 1
```
[the remark this answers is not in the document]
Comment (bastelfreak) [reply lib/puppet-lint/plugins/check_whitespace/arrow_alignment.rb:107]: looks ok to me, but I don't know enough about ruby to ensure this works. could be handled with a test.

File: lib/puppet-lint/data.rb [hunk 3]
```diff
@@ -68,10 +68,10 @@ def insert(index, token)
       token.next_token = current_token.next_token
       token.prev_token = current_token
 
-      current_token.next_token.prev_token = token unless current_token.next_token.nil?
+      current_token.next_token&.prev_token = token
```
[the remark this answers is not in the document]
Comment (bastelfreak) [reply lib/puppet-lint/data.rb:71]: I did the safe autofix with ruby 2.4 as target version

File: lib/puppet-lint/plugins/check_whitespace/arrow_alignment.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply lib/puppet-lint/plugins/check_whitespace/arrow_alignment.rb:107]: I checked and `Enumable#sum` was introduced in Ruby 2.4 so my suggestion is only valid after this PR's changes.

**Comment by chelnak** [comment]
@bastelfreak Appreciate this PR has been sat for a while now but I wonder if it's worth updating it to bump the min Puppet version to 2.5?

**Comment by chelnak** [comment]
See #50 for an updated version of this PR.

**Pull request closed without merging**